app/models/reconciliation.rb

```diff
@@ -4,23 +4,62 @@
 class Reconciliation < ApplicationRecord
   belongs_to :statement
 
+  enum update_type: %i[single also_matching_unreconciled also_matching]
+
   validates :item, :resource_type, presence: true
 
   default_scope -> { order(updated_at: :asc) }
 
   after_commit :update_statement, :create_equivalence_claim
 
+  def possibly_updated_statements
+    return Statement.where(id: statement.id) if single
+    Statement.where(id: statement.id).or(Statement.where(where_condition_matching_name))
+  end
+
   private
 
   def update_statement
-    case resource_type
-    when 'person'
-      statement.update(person_item: item)
-    when 'party'
-      statement.update(parliamentary_group_item: item)
-    when 'district'
-      statement.update(electoral_district_item: item)
-    end
+    statement.update(item_attr => item)
+    return if single
+    other_statements_for_update.update(item_attr => item)
```

Comment:
Correct me if I'm wrong, but I _think_ this means we'll only have rows in the `reconciliations` table for the statement that the user reconciled in the UI, not for all the other statements that were updated as a result of that reconciliation being created.

Since we've already had a situation where having a log of reconciliations was useful (https://github.com/mysociety/verification-pages/pull/317) I think we should ideally keep that behaviour, but I'm happy to be convinced otherwise :)

Reply:
I did think about that, and I think that since we have the `update type in the table too, the existing script for restoring reconciliations (which just does `.touch` on each to trigger `update_statement` through the `after_commit` trigger) should reproduce the same bulk reconciliation if `update_type` is `also_matching` or `also_matching_unreconciled`. So each row in `reconciliations` represents a reproducible user action, rather than an update of a stament.

So I hope that's OK, but I might well have missed something!

Reply:
Ahh, yep that makes sense now, thanks!